=== modified file 'miniupnpc/igd_desc_parse.c'
--- miniupnpc/igd_desc_parse.c	2011-08-09 11:15:32 +0000
+++ miniupnpc/igd_desc_parse.c	2011-08-09 11:21:52 +0000
@@ -84,6 +84,8 @@
 		dstmember = datas->tmp.scpdurl;
 /*	else if( !strcmp(datas->cureltname, "deviceType") )
 		dstmember = datas->devicetype_tmp;*/
+	else if( !strcmp(datas->cureltname, "friendlyName") )
+		dstmember = datas->tmp.friendlyName;
 	if(dstmember)
 	{
 		if(l>=MINIUPNPC_URL_MAXSIZE)

=== modified file 'miniupnpc/igd_desc_parse.h'
--- miniupnpc/igd_desc_parse.h	2011-08-09 11:15:32 +0000
+++ miniupnpc/igd_desc_parse.h	2011-08-09 11:22:28 +0000
@@ -18,6 +18,7 @@
 	char scpdurl[MINIUPNPC_URL_MAXSIZE];
 	char servicetype[MINIUPNPC_URL_MAXSIZE];
 	/*char devicetype[MINIUPNPC_URL_MAXSIZE];*/
+	char friendlyName[MINIUPNPC_URL_MAXSIZE];
 };
 
 struct IGDdatas {

